spec/record_serialization_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'attr_json/model'
+RSpec.describe "AttrJson::Record serialization" do
+
+  class Product
+    include AttrJson::Model
+    attr_json :name, :string, default: "A Product"
+    attr_json :cost, :string, default: "$2.25"
+    attr_json :material, :string, default: "silk"
+    attr_json :color, :string, default: "green"
+  end
+
+  class Brand
+    include AttrJson::Model
+
+    def self.build_test
+      new(products: [Product.new] )
+    end
+
+    attr_json :name, :string, default: "Some Brand"
+    attr_json :products, Product.to_type, array: true, default: []
+  end
+
+  class Department
+    include AttrJson::Model
+
+    def self.build_test
+      new(brands: [Brand.build_test]  )
+    end
+
+    attr_json :name, :string, default: "Some Department"
+    attr_json :brands, Brand.to_type, array: true, default: []
+  end
+
+
+  class Store < ActiveRecord::Base
+    include AttrJson::Record
+
+    self.table_name = "products"
+
+    attr_json :departments, Department.to_type, container_attribute: :other_attributes, array: true, default: []
+  end
+
+
+  let(:unsaved_store) { Store.new(departments: Array.new(3) { Department.build_test }) }
+
+  describe "#as_json" do
+
+    # This seems to pass, everything is properly json-ified
+    it "produces json-value-only-hash" do
+      expect(only_json_values?( unsaved_store.as_json )).to be true
+    end
```

Comment:
Passes in Rails 6.0, may not in Rails 5.2. Likely bug in Rails that was fixed in 6.0.